src/main/java/com/google/api/codegen/config/Resources.java

```diff
@@ -80,9 +82,27 @@
   public static String getEntityName(FieldSegment segment) {
     Preconditions.checkArgument(isTemplateFieldSegment(segment));
     // TODO(shinfan): Consider finding a better way to determine the name if possible.
```

Comment:
Removed. I think it is hard to do much better than using the last item plus conflict resolution, and the gain is marginal considering the user can just rename.